### PR TITLE
feat: improve typescript comments

### PR DESCRIPTION
**Pull request by thgh**
## Description

To bring Typescript support in Payload to the next level, most options should have a small description, example and link to docs. This PR is an early draft in that direction to get the stone rolling. I won't make much updates in the coming weeks, so feel free to take this over.

I had some prettier/eslint issues, so added some extra config, feel free to revert.

These seem most important to me:
- [ ] `Config` from payload/config
- [ ] `CollectionConfig` from payload/types
- [ ] `useField` from payload/components/forms
- [ ] `useForm` from payload/components/forms
- [ ] `CollectionBeforeChangeHook` from payload/types

TODO
- [ ] Check if it makes sense to reuse the same wording in docs & comments (maybe the docs allow for longer descriptions?)
- [ ] Check if it's possible to generate the type comments based on the description in the docs

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Existing test suite passes locally with my changes
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "singleQuote": true,
```
Comment (denolfe) [inline .prettierrc:2]: Was committing this file intended? We currently don't have prettier in this repo.

Comment (thgh) [reply .prettierrc:2]: Hmm, this file does not affect anything indeed. Everything from [4efa828](https://github.com/payloadcms/payload/pull/1467/commits/4efa82820806b16ffa7ab2ba84535b5499132ad1) was me trying to get the intended formatting working.

File: src/express/types.ts [hunk 2]
```diff
@@ -8,20 +8,39 @@ import { User } from '../auth/types';
...
+   * - Configuration from payload-config.ts
+   * - Mongo model for this collection
+   * - GraphQL type metadata
+   * */
   collection?: Collection;
+  /** What triggered this request */
   payloadAPI: 'REST' | 'local' | 'graphQL';
+  /** Uploaded files */
   files?: {
+    /**  */
```
[the remark this answers is not in the document]
Comment (thgh) [reply src/express/types.ts:34]: As mentioned in PR description, it's an early draft. Didn't want to pour in too much time if this isn't the preferred direction.

**Comment by denolfe** [comment]
This is really awesome @thgh ! Well done!

**Pull request merged**